src/main/java/com/gooddata/notification/Subscription.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2004-2017, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.notification;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.gooddata.md.Meta;
+import com.gooddata.util.GoodDataToStringBuilder;
+import org.springframework.web.util.UriTemplate;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Subscription for notifications
+ */
+@JsonTypeName("subscription")
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Subscription {
+
+    public static final String URI = "/gdc/projects/{project}/users/{user}/subscriptions";
+    public static final UriTemplate URI_TEMPLATE = new UriTemplate(URI);
+
+    private final List<Trigger> triggers;
+    private final TriggerCondition condition;
+    private final MessageTemplate template;
+    private final List<String> channels;
+    private final Meta meta;
+
+
+    /**
+     * Creates Subscription
+     *
+     * @param triggers triggers of subscription
+     * @param channels list of {@link ChannelConfiguration}
+     * @param condition condition under which this subscription activates
+     * @param template of message
+     * @param meta information of subscription
+     */
+    public Subscription(final List<Trigger> triggers,
+                        final List<ChannelConfiguration> channels,
+                        final TriggerCondition condition,
+                        final MessageTemplate template,
+                        final Meta meta) {
+        this(triggers,
```

Comment:
This is public constructor and it will throw NPE immediatelly if `null` channels is provided. Maybe a `notNull` check at least for that?

Reply:
done